tfjs-layers/src/layers/nlp/models/gpt2/gpt2_backbone.ts

```diff
@@ -0,0 +1,252 @@
+      name: 'position_embedding',
+    }).apply(tokenEmbedding) as SymbolicTensor;
+
+    // Sum and apply dropout to embeddings.
+    let x = add({name: 'embeddings_add'})
+      .apply([tokenEmbedding, positionEmbedding]) as SymbolicTensor;
+    x = new Dropout({rate: args.dropout, name: 'embeddings_dropout'})
+      .apply(x) as SymbolicTensor;
+
+    // Apply successive transformer decoder blocks.
+    for(let i = 0; i < args.numLayers; i++) {
+      x = new TransformerDecoder({
+        intermediateDim: args.intermediateDim,
+        numHeads: args.numHeads,
+        dropout: args.dropout,
+        layerNormEpsilon: 1e-05,
+        // TODO(pforderique): Implement gelu.
+        activation: getActivation('relu'),
+        kernelInitializer: gpt2KernelInitializer(0.02),
+        normalizeFirst: true,
+        name: `transformer_layer_${i}`,
+      }).apply(x, {decoderPaddingMask: paddingMask}) as SymbolicTensor;
+    }
+
+    const sequenceOutput = new LayerNormalization({
+      name: 'layer_norm',
+      axis: -1,
+      epsilon: 1e-05,
+      dtype: 'float32',
+    }).apply(x) as SymbolicTensor;
+
+    // Instantiate using Functional API Model constructor.
+    super({
+      inputs: [tokenIds, paddingMask],
+      outputs: sequenceOutput,
+      name: 'gpt2_backbone'
+    });
+    this.vocabularySize = args.vocabularySize;
+    this.numLayers = args.numLayers;
+    this.numHeads = args.numHeads;
+    this.hiddenDim = args.hiddenDim;
+    this.intermediateDim = args.intermediateDim;
+    this.dropout = args.dropout ?? 0.1;
+    this.maxSequenceLength = args.maxSequenceLength ?? 1024;
+  }
+
+  override getConfig(): serialization.ConfigDict {
+    const config: serialization.ConfigDict = {
+      vocabularySize: this.vocabularySize,
+      numLayers: this.numLayers,
+      numHeads: this.numHeads,
+      hiddenDim: this.hiddenDim,
+      intermediateDim: this.intermediateDim,
+      dropout: this.dropout,
+      maxSequenceLength: this.maxSequenceLength,
+    };
+    const baseConfig = super.getConfig();
+    Object.assign(config, baseConfig);
+    return config;
+  }
+
+  override get tokenEmbedding() {
+    return this.getLayer('token_embedding');
+  }
+
+  static presets(cls: serialization.SerializableConstructor<GPT2Backbone>) {
+    return {
+      gpt2_base_en: {
+        metadata: {
+          description: '12-layer GPT-2 model where case is maintained.' +
+            'Trained on WebText.',
+          params: 124439808,
+          official_name: 'GPT-2',
+          path: 'gpt2',
+          modelCard: 'https://github.com/openai/gpt-2/blob/master/model_card.md',
+        },
+        config: {
+          vocabularySize: 50257,
+          numLayers: 12,
+          numHeads: 12,
+          hiddenDim: 768,
+          intermediateDim: 3072,
+          dropout: 0.1,
+          maxSequenceLength: 1024,
+        },
+        preprocessorConfig: {},
+        weightsUrl: 'https://storage.googleapis.com/keras-nlp/models/gpt2_base_en/v1/model.h5',
```

Comment:
We'll likely need to change all the weights urls to point to our converted model weights, but we can do that later.